### ISSUE TITLE
Add a documentation

### ISSUE THREAD

**Issue by FegDotExe** [SethBling/cbscript#10]
As it stands now, the only way to get into cbscript is by reading through the provided scripts and by trial and error, at least from what I know.
It would be extremely nice if a documentation about the fundamentals of the language could be added to the repo.

**Comment by MattBehrens02** [comment]
Seconding this as it would be a huge help.
I know this would be a huge undertaking but I can really see this being a very useful tool in the realm of datapack creation.

**Comment by deli73** [comment]
strongly agreed, this needs docs of some kind. we've been wanting a tool like this for years but it's very difficult when there's no real instructions on how it works

**Comment by SethBling** [comment]
I'd love for there to be usage documentation, but I also would rather spend my time making videos. If anyone else wants to take a stab at it, I can help correct mistakes.

**Comment by deli73** [comment]
understandable! if we can get things working we might try to work on it with some friends

**Comment by MattBehrens02** [comment]
I'd be willing to pick away at things too. College does take away a lot of my free time right now but it would be a really cool project to take a part in.

**Comment by phazejeff** [comment]
> I'd love for there to be usage documentation, but I also would rather spend my time making videos. If anyone else wants to take a stab at it, I can help correct mistakes.

Maybe make a video on general usage?

**Comment by SethBling** [comment]
Yeah, I suppose that's something I could do. Show some of the features used in my scripts to give people a starting place to experiment, and mention what I'm looking for in terms of GitHub contributions.

**Comment by deli73** [comment]
yeah, that'd be helpful. we're struggling to figure out how some of this stuff works without clear docs

**Comment by ArcticYeti** [comment]
I'd love to contribute the project with docs, but unfortunately I can't due to a lack of free time as of late.

I can however suggest checking out Sphinx with the custom Furo theme for technical documentation. It's pretty easy to get the hang of and can produce some wonderful clean docs!

https://github.com/sphinx-doc/sphinx
https://github.com/pradyunsg/furo

**Comment by zipcord24** [comment]
agreed

**Comment by AdoHTQ** [comment]
Not entirely sure how this works, but you could also use a Github Wiki so that it's on the repo.

**Comment by Budmeister** [comment]
This is something I would also like to help with but probably won't have time for until after Spring graduation. 

Maybe some generative AI would be able to help in the initial stages of figuring out the syntax.

**Comment by deli73** [comment]
> Maybe some generative AI would be able to help in the initial stages of figuring out the syntax.

generative AI sounds like a good way to introduce a bunch of hard-to-locate errors into the documentation, no thanks.
